File: BE/src/main/java/com/haivn/common_api/Poster.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;

@Entity
@Table(name = "poster")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class Poster extends BaseEntity{
    @Column(name = "name")
    private String name;
    @Column(name = "file_name")
    private String fileName;
    @Column(name = "stt")
    private Short stt;
    @Column(name = "status")
    private Short status;
}

File: FE/lib/model/model_qltvn/poster.dart
class Poster {
  int? id;
  String? name;
  String? fileName;
  int? stt;
  int? status;

  Poster({this.id, this.name, this.fileName, this.stt, this.status});

  Poster.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        fileName = json['fileName'],
        stt = json['stt'],
        status = json['status'];

  Map<String, dynamic> toJson() => {
        'name': name,
        'fileName': fileName,
        'stt': stt,
        'status': status,
      };
}
